cron.php

```diff
@@ -0,0 +1,464 @@
+		$event = $this->get_event( $timestamp, $action, $instance );
+		unset( $timestamp, $action, $instance );
+
+		if ( is_array( $event ) ) {
+			$time_start = microtime( true );
+
+			// Limit how many events are processed concurrently
+			if ( ! $this->is_internal_event( $event['action'] ) && ! $this->check_lock() ) {
+				return new WP_Error( 'no-free-threads', __( 'No resources available to run this job.', 'wp-cron-control-revisited' ) );
+			}
+
+			// Prepare environment to run job
+			ignore_user_abort( true );
+			set_time_limit( $this->job_timeout_in_minutes * MINUTE_IN_SECONDS );
+			define( 'DOING_CRON', true );
+
+			// Remove the event, and reschedule if desired
+			// Follows pattern Core uses in wp-cron.php
+			if ( false !== $event['schedule'] ) {
+				$reschedule_args = array( $event['timestamp'], $event['schedule'], $event['action'], $event['args'] );
+				call_user_func_array( 'wp_reschedule_event', $reschedule_args );
+			}
+
+			wp_unschedule_event( $event['timestamp'], $event['action'], $event['args'] );
+
+			// Run the event
+			do_action_ref_array( $event['action'], $event['args'] );
+
+			// Free process for the next event
+			if ( ! $this->is_internal_event( $event['action'] ) ) {
+				$this->free_lock();
+			}
+
+			$time_end = microtime( true );
+
+			return rest_ensure_response( array(
+				'success' => true,
+				'message' => sprintf( __( 'Job with action `%1$s` and arguments `%2$s` completed in %3$d seconds.', 'wp-cron-control-revisited' ), $event['action'], serialize( $event['args'] ), $time_end - $time_start ),
+			) );
+		} else {
+			return new WP_Error( 'no-event', __( 'The specified event could not be found.', 'wp-cron-control-revisited' ) );
+		}
+	}
+
+	/**
+	 * Find an event's data using its hashed representations
+	 */
+	private function get_event( $timestamp, $action_hashed, $instance ) {
+		$events = get_option( 'cron' );
+		$event  = false;
+
+		if ( isset( $events[ $timestamp ] ) ) {
+			foreach ( $events[ $timestamp ] as $action => $action_events ) {
+				if ( hash_equals( md5( $action ), $action_hashed ) && isset( $action_events[ $instance ] ) ) {
+					$event              = $action_events[ $instance ];
+					$event['timestamp'] = $timestamp;
+					$event['action']    = $action;
+					break;
+				}
+			}
+		}
+
+		return $event;
+	}
+
+	/**
+	 * PLUGIN UTILITY METHODS
+	 */
+
+	/**
+	 * Check if request is authorized
+	 */
+	public function check_secret( $request ) {
+		$body = $request->get_json_params();
+
+		// For now, mimic original plugin's "authentication" method. This needs to be better.
+		if ( ! isset( $body['secret'] ) || ! hash_equals( $this->secret, $body['secret'] ) ) {
+			return new WP_Error( 'no-secret', __( 'Secret must be specified with all requests', 'wp-cron-control-revisited' ) );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Events that are always run, regardless of how many jobs are queued
+	 */
+	private function is_internal_event( $action ) {
```

Comment:
`priority` doesn't quite fit, because these jobs appear last in the list the endpoint returns, so their runners are scheduled after those for non-internal events.